docs/concept-exercises enhanced guidance for after.md (#1613)
exercism/v3: docs/concept-exercises.md

**Purpose:** Provide more information about the concept(s) for a student to learn from.

Once the student completes the exercise they will be shown this file, which should provide them with a summary of what the exercise aimed to teach. If the exercise introduced new syntax, syntax samples should be included.

>  This document can also link to any additional resources that might be interesting to the student in the context of the exercise, such as:

Reviewer: ```suggestion
This document can also link to any additional resources that might be interesting to the student in the context of the exercise, such as:
```

Reviewer: This gets the PR green.

> Once the student completes the exercise they will be shown this file, which should provide them with a summary of what the exercise aimed to teach. If the exercise introduced new syntax, syntax samples should be included. This document can also link to any additional resources that might be interesting to the student in the context of the exercise.
> 
> `after.md` provides an opportunity to give advice on how the features covered by a concept are used in practice.
> Among other things, you can comment on:
> 
> - Popular usages for a feature
> - Performance characteristics and memory usage
> - Alternative approaches addressed in other exercises
> - Common pitfalls in a feature's use (e.g. casual use of multiple threads)
> - Similar features with which the concept may be confused
> - Limitations on use that may catch out the unsuspecting developer
> 
> Please provide source references/links and/or justification where appropriate.
> 

Reviewer 2: Maybe we can combine this with the last line of the previous section and shorten it a bit?

```suggestion
Once the student completes the exercise they will be shown this file, which should provide them with a summary of what the exercise aimed to teach. If the exercise introduced new syntax, syntax samples should be included.

 This document can also link to any additional resources that might be interesting to the student in the context of the exercise, such as:

- Popular usages for a feature
- Performance characteristics and memory usage
- Alternative approaches addressed in other exercises
- Common pitfalls in a feature's use (e.g. casual use of multiple threads)
- Similar features with which the concept may be confused
- Limitations on use that may catch out the unsuspecting developer

Please provide source references/links and/or justification where appropriate.

```

I was also thinking that maybe we can re-order the list a bit to start with what (I think) is the best case for including more content in the after.md file:

```
- Popular usages for a feature
- Common pitfalls in a feature's use (e.g. casual use of multiple **threads)**
- Limitations on use that may catch out the unsuspecting developer
-  Alternative approaches addressed in other exercises
- Similar features with which the concept may be confused
- Performance characteristics and memory usage
```

In particular, I want to de-emphasize performance characteristics as that has the potential of maintainers stressing performance and memory over things like how to use a feature. What do you think?

Author: Definitely
